util/helm/client.go

```diff
@@ -428,10 +429,16 @@ func (c *nativeHelmChart) GetTags(chart string, noCache bool) (*TagsList, error)
 			TLSClientConfig:   tlsConf,
 			DisableKeepAlives: true,
 		}}
+
+		repoURL, err := url.Parse(c.repoURL)
+		if err != nil {
+			return nil, fmt.Errorf("error parsing repo url: %w", err)
+		}
```

Comment:
Hi @jannfis, if we want to ensure the `repoURL` is a full-qualified URL my suggestion is to change its type from `string` to `url.URL`! The reason I think we should do that is, if the `repoURL` should always be a valid one we cannot let `nativeHelmChart` be build with an invalid one.

Also today we kinda have this problem, right? If the user pass an invalid URL the request will also fail, so if the user pass something like `""`, `"//"`, `"//example"`, `"//example.com"` the `url.Parse` function will not return an error but the request will fail anyway 'cause the URL is wrong.

WDYT?

Reply:
And also, if the URL is wrong this information we're passing to "auth.StaticCredentials" will never be used because "oras" first try a "normal request" and only if the returned code is "401 Unauthorized" it'll try to authenticate and just in the final part of the process it uses the host information

Reply:
@jannfis did you have a chance to revisit here?

Reply:
@ishitasequeira amd @crenshaw-dev any thoughts here?

Reply:
Ok, I'v spotted a problem here! I found out `GetTags` method is just called when OCI is enabled, to say if it's enable argo make two conditions:
* repo.EnableOCI
* helm.IsHelmOciRepo(repo.Repo)

The second condition checks if the repo URL doesn't have the _scheme_ part set, so the repo URL will always be without scheme when calling `GetTags`! I'll fix this code now

Reply:
Done, I've pushed the fix! And also I've added a safety check 'cause get `GetTags` should only be used with OCI registries and it 's exactly what'a happening now when `repository.Service` make the checks ([source](https://github.com/argoproj/argo-cd/blob/master/reposerver/repository/repository.go#L2336)), I just made it explicit in the Helm Client!

@ishitasequeira and @jannfis, please, check the code again if possible!